# Hello Word

In [14]:
from keras.datasets import mnist
from keras import models
from keras import layers
from keras.utils import to_categorical

In [2]:
(train_images, train_labels), (test_images, test_labels) = mnist.load_data()

11493376/11490434 [==============================] - 8s 1us/step


In [3]:
train_images.shape

(60000, 28, 28)

In [7]:
len(test_labels)

10000

In [8]:
test_labels

array([7, 2, 1, ..., 4, 5, 6], dtype=uint8)

In [10]:
network = models.Sequential()

## Adding layers to the network.

In [11]:
network.add(layers.Dense(512, activation='relu', input_shape=(28 * 28,)))
network.add(layers.Dense(10, activation='softmax'))

In [12]:
network.compile(optimizer='rmsprop', loss='categorical_crossentropy', metrics=['accuracy'])

### Before training, we will preprocess our data by reshaping it into the shape that the network expects, and scaling it so that all values are in the [0, 1] interval. Previously, our training images for instance were stored in an array of shape (60000, 28, 28) of type uint8 with values in the [0, 255] interval. We transform it into a float32 array of shape (60000, 28 * 28) with values between 0 and 1.

In [13]:
train_images = train_images.reshape((60000, 28 * 28))
train_images = train_images.astype('float32') / 255

test_images = test_images.reshape((10000, 28 * 28))
test_images = test_images.astype('float32') / 255

## We also need to categorically encode the labels

In [15]:
train_labels = to_categorical(train_labels)
test_labels = to_categorical(test_labels)

## We are now ready to train our network, which in Keras is done via a call to the fit method of the network: we "fit" the model to its training data.

In [16]:
network.fit(train_images, train_labels, epochs=5, batch_size=128)

Epoch 1/5
60000/60000 [==============================] - 6s 104us/step - loss: 0.2582 - acc: 0.9258
Epoch 2/5
60000/60000 [==============================] - 5s 88us/step - loss: 0.1038 - acc: 0.9689
Epoch 3/5
60000/60000 [==============================] - 5s 91us/step - loss: 0.0686 - acc: 0.9793
Epoch 4/5
60000/60000 [==============================] - 5s 90us/step - loss: 0.0497 - acc: 0.9849
Epoch 5/5
60000/60000 [==============================] - 6s 95us/step - loss: 0.0379 - acc: 0.9887


## We quickly reach an accuracy of 0.989 (i.e. 98.9%) on the training data. Now let's check that our model performs well on the test set too:

In [17]:
test_loss, test_acc = network.evaluate(test_images, test_labels)

10000/10000 [==============================] - 1s 69us/step


In [18]:
print('Test ACC: ', test_acc)

Test ACC:  0.9797


In [19]:
print('Test LOSS: ', test_loss)

Test LOSS:  0.06738120722207241
